### ISSUE TITLE
Experiencing errors on valid CSS

### ISSUE THREAD

**Issue by samit4me** [styled-components/stylelint-processor-styled-components#4]
I am experiencing error from `stylelint` that I am not expecting. It probably has something to do with my setup, but not sure what else to try. I've had the same issue on both Win10 and macOS both with node 6.6.0.

Linting seems to work well for basic styles, example:

```
const RedBox = styled.div`
  background-color: red;
`
```

But If I use something like interpolation it seems to break, example:

```
const stylesTest = css`
  font-size: large;
`;

const Intro = styled.p`
  ${stylesTest}
`;
```

OR

```
const logoSpin = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(-360deg);
  }
`;

const Img = styled.img`
  animation: ${logoSpin} infinite 3s linear;
  height: 80px;
`;
```

I have also experienced other errors like indentation ect which I see in your test fixtures, so I am not sure why it is failing for me. To help identify what or where I have gone wrong I have created a repo base on `create-react-app`. 

Example repo: https://github.com/samit4me/create-react-app-with-styled-components

I was getting similar issues when trying to get this working with `react-boilerplate` in the [Use styled-components](https://github.com/mxstbr/react-boilerplate/pull/1110) PR.

**Comment by mxstbr** [comment]
Thank you for providing a reproduction, this is highly likely on our side – as mentioned, this is very much in alpha and we're still trying to figure out exactly what to do in which real-world cases! I'll take a look soon!

**Comment by mxstbr** [comment]
I just published v0.0.4 that should fix a lot of these errors, would you mind trying it again?

**Comment by mxstbr** [comment]
Okay, I think this is fixed in `stylelint-processor-styled-components@0.4` – can you try using that? You'll also need to set `syntax` to `"scss"` in your `.stylelintrc`!

**Comment by samit4me** [comment]
Updated both `stylelint-processor-styled-components@0.0.4` and `styled-components@1.0.9` and added `"syntax": "scss"` to `.stylelintrc` but unfortunately, I did not see any change when running `npm run lint:css`. It feels like I am missing something here and I get the feeling it will be embarrassing when I find out what, but at the moment I have no idea what it is!

Do you have this working with another project? Is it available online somewhere so I can compare my setup to see if it is something similar?

**Comment by mxstbr** [comment]
I haven't pushed it yet. Let me clone your repo and try to find the issue!

**Comment by mxstbr** [comment]
Got it, see #6! Will fix soon! (and the `Circle` testcase in #7)

**Comment by mxstbr** [comment]
Can you try again now, with the newest version of `stylelint` and the processor?

**Comment by samit4me** [comment]
Installed the latest versions which made a difference, but I think I now get more errors, see [stylelintOutput.txt](https://github.com/samit4me/create-react-app-with-styled-components/blob/master/stylelintOutput.txt).

I'm not 100% what has changed as `stylelint-processor-styled-components` version number appears to be the same `0.0.4` so unable to rollback the version. Now that stylelintOutput is the repo we should see future changes in the output.

Below is my relevant parts of `package.json`.
```
{
  "devDependencies": {
    ...
    "stylelint": "7.5.0",
    "stylelint-config-standard": "14.0.0",
    "stylelint-processor-styled-components": "0.0.4"
  },
  "dependencies": {
    ...
    "styled-components": "1.0.10"
  },
  "scripts": {
    ...
    "lint:css": "stylelint \"./src/**/*.js\" > stylelintOutput.txt"
  }
}
```

**Comment by mxstbr** [comment]
All of these except for the `Unknown word   CssSyntaxError` errors are actual linting errors and you should fix them 😉 can you show me the source of the two files that throw the CSSSyntaxError?

**Comment by samit4me** [comment]
To give some background:
_I could not get my `styled-components` to lint correctly, so I look into the source of this repository and seen which test "should" pass and which "should" fail. I then took a copy of the [test fixtures from this repository](https://github.com/styled-components/stylelint-processor-styled-components/tree/master/test/fixtures) and put them into my [mock repo](https://github.com/samit4me/create-react-app-with-styled-components). I then commented out the fixtures that "should" fail, so I could test if it was just me. When the fixtures that "should" pass actually failed, that is when I created this issue._

I have now removed all the test fixture that I copied and now only have two linting issues:
```
src/components/App/Img.js
  7:3  ×  Expected empty line before nested rule       rule-nested-empty-line-before    
 12:1  ×  Expected empty line before non-nested rule   rule-non-nested-empty-line-before

src/components/App/Intro.js
 5:3  ×  Unknown word   CssSyntaxError
```

`src/components/App/Img.js`:
```
import styled, { keyframes } from 'styled-components';

const logoSpin = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(-360deg);
  }
`;

const Img = styled.img`
  animation: ${logoSpin} infinite 3s linear;
  height: 80px;
`;

export default Img;
```

`src/components/App/Intro.js`:
```
import styled, { css } from 'styled-components';

const stylesTest = css`
  font-size: large;
`;

const Intro = styled.p`
  ${stylesTest}
`;

export default Intro;
```
^ I do realize this is strange and can be easily solved by not using the `{ css }` but I ran into this issue when trying to share styles, so this is here to demonstrate that issue. 

Thank you for all your help, it is really appreciated 👍. Also, I apologize if I am unclear in explaining the issues, I am trying my best, but it is difficult when I don't know what I am talking about 🤔

**Comment by mxstbr** [comment]
Ah, thank you, that's very helpful, now I get it! Will take a look soon!

**Comment by evenchange4** [comment]
Run into the same problem here. Any update on this issue?

**Comment by mxstbr** [comment]
None yet, sadly, haven't come around to fix this yet 😢 submitting a failing test case as a PR would be super helpful!

**Comment by ZwaarContrast** [comment]
Having the same issue as @samit4me 

A Gist can be found over here: https://gist.github.com/ZwaarContrast/c0e344b201379601b0664d60fac0d6e9

I tried to create a generic Heading component where a tag prop determines as what html element the component renders. Any ideas how to do this otherwise?

**Comment by mgtitimoli** [comment]
Hi everyone,

I came across with this issue recently, and I could isolate what is causing it, basically when there is only one interpolation within the template literal.

One example could be something like this:

```javascript
const sharedRules = css`
  font-size: large;
`;

const LargeParagraph = styled.p`
  ${sharedRules}
`;
```

Ignore the fact that I'm using `css` (it's the only way for Atom to detect that's a css construct), it also happens when using a simple template literal.

I haven't had enough time to get into the code to try to find what is causing it, but I hope knowing the pattern that triggers it can help anyone find it.

**Comment by mxstbr** [comment]
Just released v0.1.0 which should fix at least some of these issues (ref #17), can you try it with the new version?

**Comment by mgtitimoli** [comment]
It seems the error dissappeared, anyway I'm still experiencing some weird errors like this one:
![image](https://cloud.githubusercontent.com/assets/4404683/24841816/4239dedc-1d63-11e7-84ef-79ace11561fc.png)

**Comment by samit4me** [comment]
Sorry for the delay in testing. I've just updated (see below for versions) and most of the errors are gone and only left with a single error, which may actually be my fault, can you confirm @mxstbr ???



`package.json`:

```JSON
  "devDependencies": {
    "react-scripts": "0.7.0",
    "stylelint": "7.10.1",
    "stylelint-config-standard": "16.0.0",
    "stylelint-processor-styled-components": "0.1.0"
  },
  "dependencies": {
    "react": "15.3.2",
    "react-dom": "15.3.2",
    "styled-components": "1.4.5"
  },
```

---
Errors:

```
src/components/App/Img.js
 13:1  ×  Expected empty line before rule   rule-empty-line-before

src/components/App/Intro.js
 7:1  ×  Expected empty line before rule   rule-empty-line-before
```

`src/components/App/Img.js`:

```JS
import styled, { keyframes } from 'styled-components';

const logoSpin = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(-360deg);
  }
`;

const Img = styled.img`
  animation: ${logoSpin} infinite 3s linear;
  height: 80px;
`;

export default Img;
```

`src/components/App/Intro.js`:

```
import styled, { css } from 'styled-components';

const stylesTest = css`
  font-size: large;
`;

const Intro = styled.p`
  ${stylesTest}
`;

export default Intro;
```

**Comment by mkhazov** [comment]
same here: `Expected empty line before rule   rule-empty-line-before`

**Comment by mxstbr** [comment]
The newline issue is tracked in #9!
